### PR TITLE
feat(shwap): check ID equality

### PR DESCRIPTION
**Pull request by Wondertan**
We need to check that requested ID is equal to received ID.
This was the case, but implicitly in the Bitswap layer.
This change adds another explicit check for that, ensuring IDs are printed on error

### CODE REVIEW

File: share/shwap/p2p/bitswap/sample_block.go [hunk 1]
```diff
@@ -96,19 +96,28 @@ func (sb *SampleBlock) Populate(ctx context.Context, eds eds.Accessor) error {
 }
 
 func (sb *SampleBlock) UnmarshalFn(root *share.AxisRoots) UnmarshalFn {
-	return func(data []byte) error {
+	return func(cntrData, idData []byte) error {
 		if !sb.Container.IsEmpty() {
```
Comment (cristaloleg) [inline share/shwap/p2p/bitswap/sample_block.go:100]: Why it should be empty before unmarshaling? We don't want to overwrite the current data or what?

**Review by cristaloleg — APPROVED** [review]
SGTM

File: share/shwap/p2p/bitswap/sample_block.go (hunk 1 above)
Comment (Wondertan) [reply share/shwap/p2p/bitswap/sample_block.go:100]: Yes, there are possible edge cases in tests where this function is called twice for the same data

Comment (cristaloleg) [reply share/shwap/p2p/bitswap/sample_block.go:100]: Is this really a problem?

File: share/shwap/p2p/bitswap/cid.go [hunk 2]
```diff
@@ -8,18 +8,18 @@ import (
 	mh "github.com/multiformats/go-multihash"
 )
 
-// extractCID retrieves Shwap ID out of the CID.
-func extractCID(cid cid.Cid) ([]byte, error) {
+// extractFromCID retrieves Shwap ID out of the CID.
+func extractFromCID(cid cid.Cid) ([]byte, error) {
 	if err := validateCID(cid); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("invalid cid %q: %w", cid, err)
```
Comment (walldiss) [inline share/shwap/p2p/bitswap/cid.go:14]: why %q?

**Review by walldiss — APPROVED** [review]

File: share/shwap/p2p/bitswap/cid.go (hunk 2 above)
Comment (Wondertan) [reply share/shwap/p2p/bitswap/cid.go:14]: Good catch
